### PR TITLE
Implementing a GraphQL server using Prisma, SQLite, and Nest.js with Typescript

### LINKED ISSUE #2717: Implementing a GraphQL server using Prisma, SQLite, and Nest.js with Typescript
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article higher ranking.*

Implementing a GraphQL server using Prisma, SQLite, and Nest.js with Typescript

### Introduction paragraph (2-3 paragraphs):
Please write the Introductory paragraph(s), that would be included in your article. We will use this writing snippet to help us assess overall quality before approval. We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.
Take your time and write the content as you would intend to get it published. 

Nest.js is built on Node.js and Express, which is arguably the most popular way to build servers with JavaScript. Nest is a typescript-based JavaScript framework for developing and scaling backend microservices. Being statically typed with typescript, it allows you to do various JavaScript tasks as an escape hatch. This guarantees that your code is more solid. Furthermore, it makes the code easier to navigate through your code because everything is strongly typed and more scalable for large applications. This provides an out-of-the-box application architecture that enables developers and teams to construct highly testable, scalable, loosely connected, and easily maintainable applications.

Nest.js, like other frontend frameworks, is designed to deliver an Angular-like experience. As a result, when you create a new Nest.js application, you will be launched into an environment that resembles a normal front-end Angular application. The major distinction is that what you end up creating will be your backend services. You'll set up a server, link it to databases, and do all of the typical tasks associated with a back-end-based application.

The guide will focus on getting a simple server project up and running. We will use technologies such as Prisma, Nest.js, graphql, and SQLite as the project database.

### Key takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article? Use this area to get your ideas down on the bulk of your article or tutorial.

- Discuss what Nest.js entails.
- A quick glance on Grapgql.
- What are Prisma and SQLlite?
- Create a simple posts Nest.js GraphQL server-based application with all CRUD operations supported.

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

https://docs.nestjs.com/cli/overview
https://docs.nestjs.com/techniques/database
https://www.sqlite.org/index.html
https://graphql.org/learn/queries/
https://www.apollographql.com/docs/apollo-server/
https://www.prisma.io/docs/concepts/components/prisma-client/crud

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by JuliusGikonyoNyambura**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

This Fixes #2717

### CODE REVIEW

**Comment by Dawe-7** [comment]
Hi @JuliusGikonyoNyambura, I've started reviewing your article. Please don't make any changes since that might disrupt the review process.

File: content/articles/implementing-a-graphql-server-using-prisma-sqlite-and-nestjs-with-typescript/index.md [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+    PostModule,
+    GraphQLModule.forRoot({
+      typePaths: ['./**/*.graphql'],
+    }),
+  ],
+})
+export class AppModule {}
+```
+
+From above, we are exposing the _GraphQLModule_ and _PostModule_ in the module's imports array. In addition, we are also specifying the _typePaths_ for the _GraphQLModule_.
```
Comment (Dawe-7) [inline content/articles/implementing-a-graphql-server-using-prisma-sqlite-and-nestjs-with-typescript/index.md:367]:
```suggestion
In the file above, we are exposing the _GraphQLModule_ and _PostModule_ in the module's imports array. Also, we are specifying the _typePaths_ for the _GraphQLModule_.
```

File: content/articles/implementing-a-graphql-server-using-prisma-sqlite-and-nestjs-with-typescript/index.md [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+import { PostService } from './posts.service';
+import { PrismaService } from 'src/prisma.service';
+
+@Module({
+  providers: [PostResolvers, PostService, PrismaService],
+})
+export class PostModule {}
+```
+
+From above, we are simply adding the _PostResolvers_ and _PostService_ as providers.
```
Comment (Dawe-7) [inline content/articles/implementing-a-graphql-server-using-prisma-sqlite-and-nestjs-with-typescript/index.md:346]:
```suggestion
In the file above, we are simply adding the _PostResolvers_ and _PostService_ as providers.
```

File: content/articles/implementing-a-graphql-server-using-prisma-sqlite-and-nestjs-with-typescript/index.md [hunk 3]
```diff
@@ -0,0 +1,499 @@
...
+}
+```
+
+Feel free to enter the id of the post you want to delete on the id param. Then, hit the play button at the centre and observe the results on the right side.
+
+![deleting-a-post](/engineering-education/implementing-a-graphql-server-using-prisma-sqlite-and-nestjs-with-typescript/deleting-a-post.png)
+
+### Conclusion
+At this point, we have created a GraphQL API using Prisma and Nest JS consisting of all CRUD operations.
+To gain more insight knowledge concerning Prisma, Nest JS, and GraphQL, check out the following resources:
```
Comment (Dawe-7) [inline content/articles/implementing-a-graphql-server-using-prisma-sqlite-and-nestjs-with-typescript/index.md:485]:
```suggestion
To gain more insight knowledge on Prisma, Nest JS, and GraphQL, check out the following resources:
```

**Review by Dawe-7 — CHANGES_REQUESTED** [review]
@JuliusGikonyoNyambura, I noticed that some of your sentences are too long making them hard to read. Make use of the [hemingway editor](https://hemingwayapp.com/) to avoid that. I've left a few suggestions that will improve the quality of the article. Consider making the changes and we'll move on to the next process. Overall, great article.

**Comment by JuliusGikonyoNyambura** [comment]
@Dawe-7, I have added the changes.

**Comment by JuliusGikonyoNyambura** [comment]
@Dawe-7, I have added the changes.

**Review by Dawe-7 — APPROVED** [review]

**Comment by hectorkambow** [comment]
plagiarism check done

**Review by louisefindlay23 — APPROVED** [review]
Great article, @JuliusGikonyoNyambura. I especially liked your simple explanation of GraphQL and its benefits which is easy to understand for readers new to GraphQL.

Be sure to look at the edits I've made to inform future articles.

Thanks, @Dawe-7 for peer reviewing.

**Pull request merged**